Application/gui_main.py

```diff
@@ -704,51 +740,43 @@ def init_menu(self) -> None:
         closure_style_menu = line_style_menu.addMenu('School Closure Status Line Style')
 
         # COVID Styles
-        dashed_covid = QAction('Dashed', self)
-        dashed_covid.setStatusTip('Dashed line')
-        dashed_covid.triggered.connect(lambda: self.change_style('COVID', 'dashed'))
-
-        dashdot_covid = QAction('Dash-dot', self)
-        dashdot_covid.setStatusTip('Dash-dot line')
-        dashdot_covid.triggered.connect(lambda: self.change_style('COVID', 'dashdot'))
-
-        solid_covid = QAction('Solid', self)
-        solid_covid.setStatusTip('Solid line')
-        solid_covid.triggered.connect(lambda: self.change_style('COVID', 'solid'))
-
-        covid_style_menu.addActions([dashed_covid, dashdot_covid, solid_covid])
-
-        # Closure Styles
-        dashed_closure = QAction('Dashed', self)
-        dashed_closure.setStatusTip('Dashed line')
-        dashed_closure.triggered.connect(lambda: self.change_style('Closure', 'dashed'))
-
-        dashdot_closure = QAction('Dash-dot', self)
-        dashdot_closure.setStatusTip('Dash-dot line')
-        dashdot_closure.triggered.connect(lambda: self.change_style('Closure', 'dashdot'))
-
-        solid_closure = QAction('Solid', self)
-        solid_closure.setStatusTip('Solid line')
-        solid_closure.triggered.connect(lambda: self.change_style('Closure', 'solid'))
-
-        closure_style_menu.addActions([dashed_closure, dashdot_closure, solid_closure])
+        for style, description in LINE_STYLES.items():
+            action = QAction(description, covid_style_menu)
+            action.setStatusTip(description)
+            action.triggered.connect(make_function(self.plot_canvas.update_lines,
+                                                   self.plot_canvas.covid_axes, style=style))
+            covid_style_menu.addAction(action)
+
+        for style, description in LINE_STYLES.items():
+            action = QAction(description, closure_style_menu)
+            action.setStatusTip(description)
+            action.triggered.connect(make_function(self.plot_canvas.update_lines,
+                                                   self.plot_canvas.closure_axes, style=style))
+            closure_style_menu.addAction(action)
```

Comment:
Do you want to move the closure loop content into the for loop ontop?

I think it would also work wouldn't it?
Like paste Lines 751 to 755 into 749 and removing the for loop for another time, saves efficiency?